pkg/cache/memory/historical_kvcache_test.go

```diff
@@ -0,0 +1,179 @@
+package memory
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestMemoryHistoricalKeyValueCache exercises the historical key/value cache functionality.
+func TestMemoryHistoricalKeyValueCache(t *testing.T) {
+	t.Run("basic historical operations", func(t *testing.T) {
+		cache, err := NewHistoricalKeyValueCache[string](
+			WithMaxVersionAge(100),
+		)
+		require.NoError(t, err)
+
+		// Test SetVersion and GetVersion
+		err = cache.SetVersion("key", "value1", 10)
+		require.NoError(t, err)
+
+		// Test getting the latest version
+		latestVersion := cache.getLatestVersionNumber("key")
+		require.Equal(t, int64(10), latestVersion)
+
+		// Test getting the latest value
+		latestValue, isCached := cache.GetLatestVersion("key")
+		require.True(t, isCached)
+		require.Equal(t, "value1", latestValue)
+
+		// Update the latest version
+		err = cache.SetVersion("key", "value2", 20)
+		require.NoError(t, err)
+
+		// Test getting the latest version
+		latestVersion = cache.getLatestVersionNumber("key")
+		require.Equal(t, int64(20), latestVersion)
+
+		// Test getting the latest value
+		latestValue, isCached = cache.GetLatestVersion("key")
+		require.True(t, isCached)
+		require.Equal(t, "value2", latestValue)
+
+		// Test getting exact versions
+		val, isCached := cache.GetVersion("key", 10)
+		require.True(t, isCached)
+		require.Equal(t, "value1", val)
+
+		val, isCached = cache.GetVersion("key", 20)
+		require.True(t, isCached)
+		require.Equal(t, "value2", val)
+
+		// Test getting intermediate version (should return nearest lower version)
+		val, isCached = cache.GetVersion("key", 15)
+		require.True(t, isCached)
+		require.Equal(t, "value1", val)
+
+		// Test getting version before first entry
+		_, isCached = cache.GetVersion("key", 5)
+		require.False(t, isCached)
+
+		// Test getting version after last entry
+		val, isCached = cache.GetVersion("key", 25)
+		require.True(t, isCached)
+		require.Equal(t, "value2", val)
+
+		// Test getting a version for a key that isn't cached
+		_, isCached = cache.GetVersion("key2", 20)
+		require.False(t, isCached)
+	})
+
+	t.Run("historical cache ignores TTL expiration", func(t *testing.T) {
```

Comment:
Historical cache is still subject to `maxKeys` FIFO eviction policy.

I believe it's worth test covering it.